File: Server/src/main/java/com/mancersoft/litevpnserver/transport/IVpnTransport.java
package com.mancersoft.litevpnserver.transport;

import java.io.IOException;
import java.util.concurrent.CompletableFuture;

public interface IVpnTransport {

    boolean start();

    void stop();

    boolean receiveConnection(Packet packet, String vpnIpAddress);

    void sendAsync(Packet packet);

    void receive(Packet packet);

}

File: Server/src/main/kotlin/com/mancersoft/litevpnserver/transport/IVpnTransport.kt
package com.mancersoft.litevpnserver.transport

import java.util.concurrent.CompletableFuture

interface IVpnTransport {
    val isReliable: Boolean
    fun start(): CompletableFuture<Boolean>
    fun stop()
    fun sendAsync(packet: Packet)
    fun setOnMessageListener(messageListener: ((Packet) -> Unit)?)
}